### ISSUE TITLE
Is there only Multi-Head Attention? 

### ISSUE THREAD

**Issue by kaen2891** [CyberZHG/keras-transformer#11]
Hello, Thanks for your open sources. This helps great to me.

I have some questions.

In "Attention is All You Need", the Multi-Head Attention is in encoder, decoder and the Masked Multi-Head Attention is only in the decoder part. But in your transformer code, I guess there's no mask part. Is it wrong? Or if there, would you let me know the "Masked" part?

Thanks a lot.

**Comment by CyberZHG** [comment]
Enable "masked" by adding `history_only`:

https://github.com/CyberZHG/keras-transformer/blob/b5aa38e628f0f3192c1fc3e97c5746aefc695919/keras_transformer/transformer.py#L69

**Comment by kaen2891** [comment]
Thanks a lot. Now I understanding about history_only.

And... I got more questions
1.Question one is, if I want to use "attention", then should I change attention_activation change to True?

2.Second is, how can I use translation?
In "README.md", there are only same sentence. 

I think is better to show my code.

`
Purpose: translate given X to Y
X: English sample sentence
Y: German sentence
# Input part
    input_layer = keras.layers.Input(
        shape=(time_step, input_dim), 
        name='Input',
    )
# Encoder part
    ''' 
    this is your transformer's get_encoders. I want to use this for encoding
    I want to use encoder_num:6, head_num:8, hidden_dim:512, dropout_rate=0.1 
        def get_encoders(encoder_num,
                 input_layer,
                 head_num,
                 hidden_dim,
                 attention_activation=None,
                 feed_forward_activation='relu',
                 dropout_rate=0.0,
                 trainable=True):       
    '''
encoder_output = transformer.get_encoders(encoder_num=6, input_layer=input_layer, head_num=8, hidden_dim=512, dropout_rate=0.1)

# Decoder part
    ''' 
    this is your transformer's get_docoders. I want to use this for decoding
    I want to use decoder_num:6, head_num:8, hidden_dim:512, dropout_rate=0.1
    def get_decoders(decoder_num,
                 input_layer,
                 encoded_layer,
                 head_num,
                 hidden_dim,
                 attention_activation=None,
                 feed_forward_activation='relu',
                 dropout_rate=0.0,
                 trainable=True)
    '''
**decoder_output = transformer.get_decoders(decoder_num=6, input_layer=???, encoded_layer=encoder_output,
                                              head_num=8, hidden_dim=128)**

# Train part
`'''

    model = keras.models.Model(inputs=input_layer, outputs=decoder_output)
    model.compile(
        loss='mean_squared_error',
        optimizer='adam',
        metrics=["accuracy"])

    early_stopping = EarlyStopping(patience=new_args['early_patience'])
    print(model.summary())
    plot_model(model, to_file='./model_spec_transformer.png')
    hist = model.fit(X_train, Y_train, validation_data=(X_valid, Y_valid), epochs=new_args['epoch'],
                     batch_size=new_args['batch_size'], shuffle=True, callbacks=[early_stopping])
    '''
`
`
I wonder what is decoder's input_layer. Like paper(Attention is All You Need), I want to translate X->Y.
I think in decoder's Masked Multi-Head Attention Layers need encoder_output and Multi-Head Attention Layer need Maksed Multi-Head Attention Layers's Query. Would you let me know that how can I use decoder part?
I'm sorry for long questions. Thank you very much!

**Comment by CyberZHG** [comment]
`attention_activation ` could be an identifier (e.g. `'relu'`) or an activation function (e.g. 
 `keras.activations.relu`).

**Comment by CyberZHG** [comment]
A simple translation example has been added to the end of README.
